use-cases/flight-crew-scheduling/src/main/java/org/acme/flighcrewscheduling/domain/Employee.java

```diff
@@ -0,0 +1,174 @@
+package org.acme.flighcrewscheduling.domain;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import ai.timefold.solver.core.api.domain.entity.PlanningEntity;
+import ai.timefold.solver.core.api.domain.lookup.PlanningId;
+import ai.timefold.solver.core.api.domain.variable.InverseRelationShadowVariable;
+
+import com.fasterxml.jackson.annotation.JsonIdentityInfo;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+
+@PlanningEntity
+@JsonIdentityInfo(scope = Employee.class, generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
+public class Employee {
+
+    @PlanningId
+    private String id;
+    private String name;
+    private Airport homeAirport;
+
+    private List<String> skills;
+    private List<LocalDate> unavailableDays;
+
+    @JsonIgnore
+    @InverseRelationShadowVariable(sourceVariableName = "employee")
+    private SortedSet<FlightAssignment> flightAssignments;
```

Comment:
Hmm, I see. I wouldn't sort it in the `countInvalidConnections` method, because that will significantly slow down the constraints. That's probably the reason why it's done this way. 

Maybe an explanatory comment?
Or maybe - since this is a quickstart - we ignore the performance issue?

Reply:
However, I think sorting the flights for every `countInvalidConnections` call may impact performance. Should we replace it with a `List`?

Reply:
This is actually a great example of an anti-pattern. We're doing something iteratively which we should have expressed as a constraint.

Here's an exercise for you: rack your brain for a while, figure out how to do the constraint without relying on the method. :-) It shouldn't be too hard.